sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/StoredProcedureRetryThrottleTest.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.models.PartitionKey;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Mockito.doAnswer;
+
+public class StoredProcedureRetryThrottleTest extends TestSuiteBase {
+    private final static int TIMEOUT = 10000;
+
+    private SpyClientUnderTestFactory.ClientWithGatewaySpy client;
+    private DocumentCollection createdCollection;
+
+    public StoredProcedureRetryThrottleTest() {}
+
+    @Test(groups = { "long" }, timeOut = TIMEOUT)
```

Comment:
Can we put this test in "emulator" group ?

Reply:
changed to emulator group. can I know what is the concern here?

Reply:
Of course, since this test is not a very big test and uses mocking, so it won't take much resources. 
Running it in "emulator" group makes sure it runs in the small CI that automatically triggers when we create the PR. 
Whereas if you run the test in "long" group, then you will have to trigger full live CI to run this test case - which doesn't make sense for these quick test cases.

Reply:
got it, thanks for the explanation~